### PR TITLE
#195: Option to enable case insensitive wildcard mapping

### PR DESCRIPTION
**Pull request by grubeninspekteur**
## Issue link
#195 Dozer partially case sensitive in >=5.5.1

## Purpose
Adds an option "wildcard-case-insensitive" to ignore the case of fields in wildcard mappings. The behavior is undefined if a single source or target class has multiple fields that only differ in case.

## Open Questions and Pre-Merge TODOs
- [ ] Issue created
- [ ] Unit tests pass
- [ ] Documentation updated
- [ ] Travis build passed

### CODE REVIEW

**Comment by grubeninspekteur** [comment]
Don't know why the OSGI tests fail on Travis, they run fine locally.

**Comment by garethahealy** [comment]
OSGi due to a test flake.

@orange-buffalo ; can you review? i'll try to review sometime this week.

File: docs/asciidoc/documentation/globalConfiguration.adoc [hunk 1]
```diff
@@ -45,9 +45,34 @@ For example, the following mapping does not allow wildcards:
 </mapping>
 ----
 
+Per default, wildcard mappings are case sensitive. To change this, you
+can enable the wildcard case insensitive policy, either globally or
+at the mapping level:
+
+[source,xml,prettyprint]
+----
+<mapping wildcard-case-insensitive="true">
```
Comment (orange-buffalo) [inline docs/asciidoc/documentation/globalConfiguration.adoc]: I would suggest to demonstrate the relevant part of `<configuration>` block here, as this page is about the global config.

Comment (grubeninspekteur) [reply docs/asciidoc/documentation/globalConfiguration.adoc]: Actually the entire page suffers from this problem: https://dozermapper.github.io/gitbook/documentation/globalConfiguration.html

Shall I rewrite the other sections as well?

Comment (orange-buffalo) [reply docs/asciidoc/documentation/globalConfiguration.adoc]: I would appreciate that!

**Pull request merged**